lib/python-caller-trampoline.py

```diff
@@ -164,7 +164,13 @@ def worker_loop():
 def terminate_worker(signum, stack):
     if worker_pid > 0:
         os.kill(worker_pid, signal.SIGKILL)
-    sys.exit(0)
+    try:
+        sys.exit(0)
+    except BaseException as e:
```

Comment:
Given that `sys.exit(0)` raises a `SystemExit` exception, and that [`SystemExit` extends from `BaseException`](https://docs.python.org/3/library/exceptions.html#SystemExit), won't we hit the `except` case 100% of the time?

Reply:
Then could `os._exit(0)` be used right away, or instead, add an initial handler to re-raise the `SystemExit` case? Like this:

``` python
try:
    sys.exit(0)
except SystemExit as e:
    raise e
except BaseException as e:
    # ...
```

I don't know if re-raising is proper for this, though.

Reply:
On second thought, my second suggestion doesn't help if the exit call does raise a failure exception, since it will not be handled by the other case when re-raised. So, what about resorting to `os._exit(0)` initially?

Reply:
I considered using `os._exit(0)` originally, but I thought that `sys.exit(0)` would probably clean up some relevant data. If that is not necessary (considering that #3731 replaced it in other points of the code), I'm happy to change it to use that instead. @nwalters512 can you think of any reason why keeping `sys.exit(0)` would be needed in this case? Maybe a parent process that needs to handle the child process?

Reply:
Yeah, I was thinking about that. We should only hit this case if either a) the parent is specifically trying to kill the zygote, or b) if the user sends SIGINT, and in both cases we shouldn't have to worry about cleanup, since zygotes are meant to be stateless and free of side effects. I think that should be safe.